src/Marten/QueryableExtensions.cs

```diff
@@ -492,43 +555,74 @@ public static IOrderedQueryable<T> ThenBy<T>(this IOrderedQueryable<T> queryable
         /// <returns></returns>
         public static IOrderedQueryable<T> ThenByDescending<T>(this IOrderedQueryable<T> queryable, string property)
         {
-            return ApplyOrder<T>(queryable, property, "ThenByDescending");
+            return ApplyOrder(queryable, property, "ThenByDescending");
+        }
+
+        /// <summary>
+        /// Chain another order by using a single property in descending order
+        /// </summary>
+        /// <param name="queryable"></param>
+        /// <param name="property"></param>
+        /// <typeparam name="T"></typeparam>
+        /// <returns></returns>
+        public static IBatchedOrderedQueryable<T> ThenByDescending<T>(this IBatchedOrderedQueryable<T> queryable, string property)
+        {
+            return ApplyOrder(queryable, property, "ThenByDescending");
         }
 
         private static IOrderedQueryable<T> ApplyOrder<T>(
             IQueryable<T> queryable,
             string property,
             string methodName)
         {
+            var lambda = CompileOrderBy<T>(property, out var targetType);
+            var result = typeof(Queryable).GetMethods().Single(
```

Comment:
**COMMENT:** Not for this PR, but some memoisation will be good to add here.

Reply:
We will add a TODO comment for this, will merge the PR.